### PR TITLE
Prompt overwrite file message

### LINKED ISSUE #2295: Overwrite prompt missing.
Please add something to the effect of

`Are you sure you want to override "preciousbits.ivf"? [y/N]`

I just got burned by the lack of this little tidbit.
Thanks.

### PR DESCRIPTION
**Pull request by Dayof**
PR to fix #2295 :
- Add function ``IvfMuxer::check_file(path)``
- Calling from ``bin/muxer/mod.rs``, ``create_muxer(path)``
- Default value "N" 
- Add arg ``-y`` to overwrite uninteractive

### CODE REVIEW

**Comment by rzumer** [comment]
The implementation looks good to me, but it needs a new CLI option that will bypass the check and always overwrite, for applications that don't want to rely on user input.

**Comment by Dayof** [comment]
@rzumer do you think is good to merge the cli addition in the PR ?

**Comment by rzumer** [comment]
Yes, you might want to make it a separate commit but it's definitely better to have it merged at the same time.

File: src/bin/muxer/ivf.rs [hunk 1]
```diff
@@ -45,6 +46,27 @@ impl Muxer for IvfMuxer {
 }
 
 impl IvfMuxer {
+  pub fn check_file(path: &str) -> Result<(), CliError> {
+    if fs::metadata(path).is_ok() {
+      print!("File '{}' already exists. Overwrite ? [y/N] ", path);
```
Comment (lu-zero) [inline src/bin/muxer/ivf.rs:51]: `eprint` and `eprintln` below might be better.

File: src/bin/muxer/ivf.rs [hunk 2]
```diff
@@ -45,6 +46,27 @@ impl Muxer for IvfMuxer {
...
+      let mut option_input = String::new();
+      io::stdin()
+        .read_line(&mut option_input)
+        .expect("Failed to read option.");
+
+      match option_input.as_str().trim() {
+        "y" | "Y" => return Ok(()),
+        _ => {
+          println!("Not overwriting, exiting.");
+          std::process::exit(0);
```
Comment (lu-zero) [inline src/bin/muxer/ivf.rs:63]: You should return a CliError and forward it up.

Comment (Dayof) [reply src/bin/muxer/ivf.rs:63]: Sorry for the silly question, but instead of what ? ``std::process::exit(0);`` ?

Comment (lu-zero) [reply src/bin/muxer/ivf.rs:63]:
``` rust
return Err(CliError::NotOverwriting)
```

**Pull request merged**